server/handlers.go

```diff
@@ -716,6 +721,27 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 		return
 	}
 
+	// Check for code challenge validity
+	if authCode.CodeChallenge != "" {
+		if verifier == "" {
+			s.tokenErrHelper(w, errAccessDenied, "Require code_verifier", http.StatusUnauthorized)
+			return
+		}
+		var challenge string
+		switch strings.ToLower(authCode.CodeChallengeMethod) {
```

Comment:
client would lookup the discovery to know the supported challenge methods which are "plain" and "S256" in this case. case conversion not necessary.